### PR TITLE
Promote exceed quota e2e test for Replication Controller to conformance test

### PR DESCRIPTION
**Pull request by globervinodhn**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
 /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**: This PR requests to promote an existing E2E test for Replication Controller to conformance. The E2E test for a common issue in Replication Controller which involves verifying that the RS creation fails when the request for the number of pods exceeds the allocated pod count in the concerned namespace.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**: The test is non-flaky, has no node dependency and goes on to verify the expected behaviour of the Replication Controller. The E2E tests take approximately 8 seconds to execute

**Does this PR introduce a user-facing change?**: NONE

release-note-none

cc @brahmaroutu @mgdevstack 
/area conformance

### CODE REVIEW

**Review by timothysc — APPROVED** [review]
/approve 
/lgtm 
/hold 

I don't see why this shouldn't be promoted
@bgrant0607 @spiffxp @dims @smarterclayton  

For final approval.

**Comment by dims** [comment]
@mattfarina @prydonius @kow3ns is this ok from the sig-apps perspective? 

Since the OWNERS file indicates this test belongs to sig-apps (https://github.com/kubernetes/kubernetes/blob/master/test/e2e/apps/OWNERS#L9-L11)

**Comment by dims** [comment]
@globervinodhn please add a release note and get the CI job to go green as well.

**Comment by spiffxp** [comment]
Looks like you manually added the test, please follow the instructions in test/conformance/README.md
```
--- test/conformance/testdata/conformance.txt	2019-03-06 02:18:30.160808395 +0000
+++ test/conformance/conformance.txt	2019-03-06 02:24:06.741681352 +0000
@@ -24,9 +24,9 @@
 test/e2e/apps/deployment.go: "deployment should support rollover"
 test/e2e/apps/deployment.go: "deployment should support proportional scaling"
 test/e2e/apps/rc.go: "should serve a basic image on each replica with a public image"
+test/e2e/apps/rc.go: "should surface a failure condition on a common issue like exceeded quota"
 test/e2e/apps/rc.go: "should adopt matching pods on creation"
 test/e2e/apps/rc.go: "should release no longer matching pods"
-test/e2e/apps/rc.go: "should surface a failure condition on a common issue like exceeded quota"
 test/e2e/apps/replica_set.go: "should serve a basic image on each replica with a public image"
 test/e2e/apps/replica_set.go: "should adopt matching pods on creation and release no longer matching pods"
 test/e2e/apps/statefulset.go: "should perform rolling updates and roll backs of template modifications"
See instructions in test/conformance/README.md
```

File: test/conformance/testdata/conformance.txt [hunk 1]
```diff
@@ -26,6 +26,7 @@ test/e2e/apps/deployment.go: "deployment should support proportional scaling"
 test/e2e/apps/rc.go: "should serve a basic image on each replica with a public image"
 test/e2e/apps/rc.go: "should adopt matching pods on creation"
 test/e2e/apps/rc.go: "should release no longer matching pods"
+test/e2e/apps/rc.go: "should surface a failure condition on a common issue like exceeded quota"
```
Comment (spiffxp) [inline test/conformance/testdata/conformance.txt]: this was added by hand so it's failing the check, please see the diff to know where to put it, or follow the instructions I linked to

**Review by spiffxp — CHANGES_REQUESTED** [review]
need to address the test list diff

File: test/conformance/testdata/conformance.txt (hunk 1 above)
Comment (globervinodhn) [reply test/conformance/testdata/conformance.txt]: Thanks for the review. Commit rebuilt and refreshed as per the expected norm

**Comment by brahmaroutu** [comment]
@smarterclayton @enisoc   Can you please review and lgtm to promote this PR.

File: test/e2e/apps/rc.go [hunk 2]
```diff
@@ -56,7 +56,12 @@ var _ = SIGDescribe("ReplicationController", func() {
 		TestReplicationControllerServeImageOrFail(f, "private", imageutils.GetE2EImage(privateimage))
 	})
 
-	It("should surface a failure condition on a common issue like exceeded quota", func() {
+	/*
+		Release : v1.14
```
Comment (timothysc) [inline test/e2e/apps/rc.go]: v1.15

**Review by timothysc — APPROVED** [review]
/approve 
/hold 
/assign @bgrant0607 @smarterclayton 

Please adjust your approved PRs for v1.15

**Review by spiffxp — APPROVED** [review]

**Comment by bclau** [comment]
FWIW, the test passes on Windows.

**Comment by oomichi** [comment]
This seems to fit for the requirements(https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md#conformance-test-requirements)

/lgtm

**Pull request merged**